OE-Core uses "make modules && make modules install" for kernel modules
It also uses KERNEL_SRC to specify the kernel source tree location

This patch adds support for modules and modules_install targets, and
adds logic to pickup INSTALL_MOD_PATH and use KERNEL_SRC

Upstream-status: Pending

diff -Nuarp mcc-kmod-1.03-orig/Makefile mcc-kmod-1.03/Makefile
--- mcc-kmod-1.03-orig/Makefile	2013-08-12 14:11:56.439711957 -0400
+++ mcc-kmod-1.03/Makefile	2013-08-12 14:16:03.472603053 -0400
@@ -3,17 +3,19 @@ mcc-y = mcc_linux.o mcc_shm_linux.o mcc_
 
 PWD := $(shell pwd)
 
-EXTRA_CFLAGS += -I$(KERNELDIR)/include -Wno-format
+EXTRA_CFLAGS += -I$(KERNEL_SRC)/include -Wno-format
 
-all:
-	$(MAKE) -C $(KERNELDIR) M=$(PWD) modules
+DESTDIR ?= $(INSTALL_MOD_PATH)
 
-install: all
-	$(MAKE) -C $(KERNELDIR) M=$(PWD) modules_install INSTALL_MOD_PATH=$(DESTDIR)
+modules all:
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) modules
+
+modules_install install: all
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) modules_install INSTALL_MOD_PATH=$(DESTDIR)
 	@echo Copying mcc headers to toolchain
 	mkdir -p $(DESTDIR)/usr/include/linux
 	cp -f {mcc_linux.h,mcc_common.h,mcc_config.h} $(DESTDIR)/usr/include/linux/
 
 
 clean:
-	$(MAKE) -C $(KERNELDIR) M=$(PWD) clean
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) clean
